Terminal.Gui/Views/Button.cs

```diff
@@ -104,39 +132,49 @@ public override bool WantContinuousButtonPressed
         }
     }
 
-    private void Button_MouseClick (object sender, MouseEventEventArgs e)
-    {
-        e.Handled = InvokeCommand (Command.HotKey) == true;
-    }
+    private void Button_MouseClick (object sender, MouseEventEventArgs e) { e.Handled = InvokeCommand (Command.HotKey) == true; }
 
     private void Button_TitleChanged (object sender, EventArgs<string> e)
     {
         base.Text = e.CurrentValue;
         TextFormatter.HotKeySpecifier = HotKeySpecifier;
     }
 
-    /// <inheritdoc />
+    /// <inheritdoc/>
     public override string Text
     {
-        get => base.Title;
-        set => base.Text = base.Title = value;
+        get => Title;
+        set => base.Text = Title = value;
     }
 
-    /// <inheritdoc />
+    /// <inheritdoc/>
     public override Rune HotKeySpecifier
     {
         get => base.HotKeySpecifier;
         set => TextFormatter.HotKeySpecifier = base.HotKeySpecifier = value;
     }
 
-    /// <summary>Gets or sets whether the <see cref="Button"/> is the default action to activate in a dialog.</summary>
-    /// <value><c>true</c> if is default; otherwise, <c>false</c>.</value>
+    /// <summary>
+    ///     Gets or sets whether the <see cref="Button"/> will show an indicator indicating it is the default Button. If
+    ///     <see langword="true"/>
+    ///     <see cref="Command.Accept"/> will be invoked when the user presses <c>Enter</c> and no other peer-
+    ///     <see cref="View"/> processes the key.
+    ///     If <see cref="View.Accept"/> is not handled, the Gets or sets whether the <see cref="Button"/> will show an
+    ///     indicator indicating it is the default Button. If <see langword="true"/>
+    ///     <see cref="Command.Accept"/> command on the <see cref="View.SuperView"/> will be invoked.
+    /// </summary>
```

Comment:
Comment very unclear? copy paste error?

Also don't lead with describing it as showing an indicator.

The indicator is almost incidental - thr important bit to user is that it makes it fire on enter even when not focused

Reply:
`e.Handled`, `e.Cancel` + APIs that return `bool?` can be very confusing. Multiple times I've introduced bugs because I got the logic wrong. I've tried to add API docs (even to `private` methods) to try to help with this. 

Here's the equivalent from `View.RaiseSelectEvent`:

> `true` the event was canceled. If `false` the event was raised but not canceled. If `null`" no event was raised.

I'd like help in phraseology here.

Specifically, regarding `Button`: In `v1`, it was very closely coupled with `View` and `Dialog`. In earlier v2 it was also closely coupled with `Window`. Now that we have consistent `Accept` support across `View`, it is ALMOST completely decoupled (the only remaining coupling is `Button` is the only `View` that has an explicit `IsDefault` property.

In this PR we have:

* All Views (and `Button` with `IsDefault == false`)
  * `Command.Accept` raises `Accept/OnAccept`. 
  * If the event is handled elsewhere, processing stops. 
  * If not handled, we look for a peer `Button` with `IsDefault = true` and invoke `Command.Accept` on it. 
  * If there's no peer-`Buttton.IsDefault` we invoke `Command.Accept` on the `SuperView`.

This enables `Accept` (via `Key.Enter` by default) to bubble up the Superview hierarchy. `Dialog`, for example, can now put it's "close" handler on `this.Accept` instead of having to do gymnastics around a `Button` that may or may not be added. It means ANY SuperView can handle `this.Accept`.  Run the `Text Input Controls` scenario and press `Enter` while on the `TextField` or change the `TextView` to `!Multiline` and you'll see this:

![JJ6uuDu 1](https://github.com/user-attachments/assets/6e5c6cef-230b-4f7a-ada2-7ea37ac9d6bb)

This is wonderful for most keyboard interaction. But what about mouse? What about when the user presses the button's HotKey when it's not focused? What if the user clicks on a `Button is { IsDefault: True}`? 

What they expect is the `Superview`'s `Accept` event will be fired. 

The only way I was able to address this was to special case `Button` such that when clicked it checks `IsDefault` and if set, invokes `SuperView`'s `Command.Accept`.

Open to other ideas for this. In the meantime I'll clear up the API docs the best I can.